sympy/polys/polytools.py

```diff
@@ -6298,7 +6298,7 @@ def factor(f, *gens, **args):
     >>> factor((x**2 - 1)/(x**2 + 4*x + 4))
     (x - 1)*(x + 1)/(x + 2)**2
     >>> factor((x**2 + 4*x + 4)**10000000*(x**2 + 1))
-    (x + 2)**20000000*(x**2 + 1)
+    (x + 2)**20*10**6*(x**2 + 1)
```

Comment:
I think we need the printer to put parenthesis around this.

Reply:
IMHO, putting parenthesis may invite more bugs and the precedence of mathematical operators results in only one interpretation of the above output, so the APIs are still unambiguous. But still if still parenthesis are required, let me know.

Reply:
In this case it's strictly needed (due to operator precedence):
```
In [1]: x**20*10**6                                                             
Out[1]: 
         20
1000000⋅x  

In [2]: x**(20*10**6)                                                           
Out[2]: 
 20000000
x  
```